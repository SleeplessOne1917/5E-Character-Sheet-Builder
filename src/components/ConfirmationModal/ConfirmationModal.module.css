.modal {
    width: 100%;
    height: 100%;
    background-color: var(--heaven);
    display: grid;
    grid-template-columns: auto;
    grid-template-rows: 5fr 1fr;
    grid-template-areas: "content"
                          "buttons";
    font-family: var(--font-display);
    overflow: hidden;
}

.content {
    grid-area: content;
    display: flex;
    flex-direction: column;
    align-items: center;
    text-align: center;
    justify-content: center;
    font-size: 1.5rem;
    padding: 0.5rem;
    overflow-y: auto;
}

.buttons {
    grid-area: buttons;
    display: flex;
}

.icon {
    fill: var(--highlight);
    stroke: var(--davy);
    width: 10rem;
}

@media screen and (min-width: 768px) {
    .modal {
        height: 40%;
        width: 50%;
    }
}

@media screen and (min-width:1024px) {
    .modal {
        width: 35%;
    }
}