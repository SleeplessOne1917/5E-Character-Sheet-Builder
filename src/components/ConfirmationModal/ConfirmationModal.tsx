import { CSSProperties } from 'react';
import Button from '../Button/Button';
import classes from './ConfirmationModal.module.css';
import { ExclamationIcon } from '@heroicons/react/outline';

type ConfirmationModalProps = {
	show: boolean;
	onYes: () => void;
	onNo: () => void;
	message: string;
};

const ConfirmationModal = ({
	show,
	onYes,
	onNo,
	message
}: ConfirmationModalProps): JSX.Element => {
	const buttonStyle: CSSProperties = {
		flexGrow: 1,
		display: 'flex',
		justifyContent: 'center',
		alignItems: 'center',
		fontSize: '2rem',
		border: 0,
		margin: 0,
		borderRadius: 0
	};

	return (
		<div
			className={classes['modal-container']}
			data-testid="confirmation-modal"
			style={{ display: show ? 'flex' : 'none' }}
		>
			<div className={classes.modal}>
				<div className={classes.content}>
					<ExclamationIcon className={classes.icon} /> {message}
				</div>
				<div className={classes.buttons}>
					<Button positive style={buttonStyle} onClick={onYes}>
						Yes
					</Button>
					<Button style={buttonStyle} onClick={onNo}>
						No
					</Button>
				</div>
			</div>
		</div>
	);
};

export default ConfirmationModal;
