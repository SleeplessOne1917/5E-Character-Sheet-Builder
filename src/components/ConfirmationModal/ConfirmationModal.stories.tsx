import { ComponentMeta, ComponentStory } from '@storybook/react';
import ConfirmationModal from './ConfirmationModal';

export default {
	title: 'Components/ConfirmationModal',
	component: ConfirmationModal,
	args: {
		show: true,
		message: 'Are you sure you want to do that?'
	}
} as ComponentMeta<typeof ConfirmationModal>;

const Template: ComponentStory<typeof ConfirmationModal> = args => (
	<ConfirmationModal {...args} />
);

export const Default = Template.bind({});
