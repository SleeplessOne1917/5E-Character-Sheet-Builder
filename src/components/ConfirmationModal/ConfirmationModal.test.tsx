import { render, screen } from '@testing-library/react';
import * as stories from './ConfirmationModal.stories';
import { composeStories } from '@storybook/react';

const { Default } = composeStories(stories);

it('renders correctly', () => {
	render(<Default />);

	expect(screen.getByTestId('confirmation-modal')).toMatchSnapshot();
});
