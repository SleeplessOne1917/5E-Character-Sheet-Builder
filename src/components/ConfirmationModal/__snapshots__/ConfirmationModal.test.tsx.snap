// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="modal-container"
  data-testid="confirmation-modal"
  style="display: flex;"
>
  <div
    class="modal"
  >
    <div
      class="content"
    >
      <svg
        aria-hidden="true"
        class="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 10.5v3.75m-9.303 3.376C1.83 19.126 2.914 21 4.645 21h14.71c1.73 0 2.813-1.874 1.948-3.374L13.949 4.88c-.866-1.501-3.032-1.501-3.898 0L2.697 17.626zM12 17.25h.007v.008H12v-.008z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
       
      Are you sure you want to do that?
    </div>
    <div
      class="buttons"
    >
      <button
        class="button positive"
        style="font-size: 2rem; padding: 1rem; margin: 0px; flex-grow: 1; display: flex; justify-content: center; align-items: center; border: 0px; border-radius: 0;"
        type="button"
      >
        Yes
      </button>
      <button
        class="button negative"
        style="font-size: 2rem; padding: 1rem; margin: 0px; flex-grow: 1; display: flex; justify-content: center; align-items: center; border: 0px; border-radius: 0;"
        type="button"
      >
        No
      </button>
    </div>
  </div>
</div>
`;
